Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -50,10 +50,14 @@ open class _ChatChannelListVC<ExtraData: ExtraDataTypes>: _ViewController,
         controller.synchronize()
         
         collectionView.register(uiConfig.channelList.channelViewCell.self, forCellWithReuseIdentifier: "Cell")
-        collectionViewLayout.register(
-            uiConfig.channelList.channelCellSeparatorReusableView,
-            forDecorationViewOfKind: collectionViewLayout.separatorIdentifier
-        )
+
+        if let cellSeparatorIdentifier = (collectionViewLayout as? CellSeparatorContainingLayout)?.separatorIdentifier {
+            collectionViewLayout.register(
+                uiConfig.channelList.channelCellSeparatorReusableView,
+                forDecorationViewOfKind: cellSeparatorIdentifier
+            )
+        }
```

Comment:
Does the `collectionViewLayout` property really need to be `UICollectionViewLayout`? I thought we should use the views of UIKit, only if we didn't need any specific data. 

Another problem with this is that we will be repeating this a lot. There are other views that require the cell separator, and it feels we making this less reusable and less obvious to reuse 🤔